File: festival-web/src/main/java/cn/bdqfork/web/route/filter/Filter.java
package cn.bdqfork.web.route.filter;


import io.vertx.ext.web.RoutingContext;

/**
 * @author bdq
 * @since 2020/1/28
 */
@FunctionalInterface
public interface Filter {

    void doFilter(RoutingContext routingContext, FilterChain filterChain) throws Exception;

}

File: festival-kotlin-web/src/main/java/cn/bdqfork/web/route/filter/Filter.kt
package cn.bdqfork.web.route.filter

import io.vertx.ext.web.RoutingContext

/**
 * @author bdq
 * @since 2020/1/28
 */
@FunctionalInterface
interface Filter {
    @Throws(Exception::class)
    fun doFilter(routingContext: RoutingContext, filterChain: FilterChain)
}